domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs

```diff
@@ -51,65 +56,48 @@ public async Task<ActionResult> CreateTransferAgreementProposal([FromQuery] Guid
             return ValidationProblem(ModelState);
         }
 
-        var newProposal = new TransferAgreementProposal
-        {
-            SenderCompanyId = OrganizationId.Create(organizationId),
-            SenderCompanyTin = Tin.Create(identityDescriptor.OrganizationCvr!),
-            SenderCompanyName = OrganizationName.Create(identityDescriptor.OrganizationName),
-            Id = Guid.NewGuid(),
-            ReceiverCompanyTin = String.IsNullOrWhiteSpace(request.ReceiverTin) ? null : Tin.Create(request.ReceiverTin),
-            StartDate = UnixTimestamp.Create(request.StartDate),
-            EndDate = request.EndDate == null ? null : UnixTimestamp.Create(request.EndDate.Value)
-        };
-
-        if (request.ReceiverTin != null)
-        {
-            if (request.ReceiverTin.Equals(identityDescriptor.OrganizationCvr))
-            {
-                return ValidationProblem(
-                    "ReceiverTin cannot be the same as SenderTin.",
-                    statusCode: 400);
-            }
-
-            var hasConflict = await unitOfWork.TransferAgreementRepo.HasDateOverlap(newProposal);
-            if (hasConflict)
-            {
-                return ValidationProblem(
-                    "There is already a Transfer Agreement with this company tin within the selected date range",
-                    statusCode: 409);
-            }
-        }
-
-        await unitOfWork.TransferAgreementProposalRepo.AddTransferAgreementProposal(newProposal);
-
-        await AppendToActivityLog(identityDescriptor, newProposal, ActivityLogEntry.ActionTypeEnum.Created);
-
-        await unitOfWork.SaveAsync();
+        var senderTin = IsOwnOrganization(organizationId) ? identityDescriptor.OrganizationCvr : null;
+        var senderName = IsOwnOrganization(organizationId) ? identityDescriptor.OrganizationName : null;
```

Comment:
Maybe we should take name from frontend?

Reply:
Permanent fix in #3792